pkg/health/registry.go

```diff
@@ -0,0 +1,56 @@
+/*
+ *    Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package health
+
+import "sync"
+
+// NewDefaultRegistry returns a default health registry with a single ping indicator and a default aggregation policy
+func NewDefaultRegistry() Registry {
+	return &defaultRegistry{
+		indicators:        []Indicator{&pingIndicator{}},
+		aggregationPolicy: &DefaultAggregationPolicy{},
+		indicatorsMutex:   sync.Mutex{},
+		aggregatorMutex:   sync.Mutex{},
+	}
+}
+
+type defaultRegistry struct {
+	indicators        []Indicator
+	aggregationPolicy AggregationPolicy
+	indicatorsMutex   sync.Mutex
+	aggregatorMutex   sync.Mutex
+}
+
+func (p *defaultRegistry) RegisterHealthAggregationPolicy(aggregator AggregationPolicy) {
+	p.aggregatorMutex.Lock()
```

Comment:
do we actually need this locking? we don't have post-start hooks so

Reply:
I thought a goroutine might register a health indicator, but then it won't be taken by the controller, so these are unnecessary